src/node-file-trace.ts

```diff
@@ -366,23 +366,19 @@ export class Job {
         await this.emitFile(pjsonBoundary + sep + 'package.json', 'resolve', path);
     }
 
-    let analyzeResult: AnalyzeResult;
+    let analyzeResultPromise: Promise<AnalyzeResult>;
 
     const cachedAnalysis = this.analysisCache.get(path);
     if (cachedAnalysis) {
-      analyzeResult = cachedAnalysis;
+      analyzeResultPromise = cachedAnalysis;
     }
     else {
-      const source = await this.readFile(path);
-      if (source === null) throw new Error('File ' + path + ' does not exist.');
-      // analyze should not have any side-effects e.g. calling `job.emitFile` 
-      // directly as this will not be included in the cachedAnalysis and won't
-      // be emit for successive runs that leverage the cache
-      analyzeResult = await analyze(path, source.toString(), this);
-      this.analysisCache.set(path, analyzeResult);
```

Comment:
Lets move this code back to how it was and just remove the `await` (no need for a new runAnalyze function)